src/test/java/it/pagopa/interop/probing/eservice/operations/unit/controller/ProducerControllerTest.java

```diff
@@ -33,40 +32,58 @@ class ProducerControllerTest {
       throws Exception {
-    Producer producer = Producer.builder().build();
 
-    producerExpectedList = List.of(producer);
-    Mockito.when(service.getEservicesProducers(2, 0, "ProducerName-Test"))
+    producerExpectedList =
+        SearchProducerNameResponse.builder().content(List.of("ProducerName-Test")).build();
+
+    Mockito.when(service.getEservicesProducers(2, 0, "ProducerName"))
         .thenReturn(producerExpectedList);
     MockHttpServletResponse response = mockMvc
         .perform(get(apiGetEservicesProducersUrl)
-            .params(getMockRequestParamsGetEservicesProducers("2", "0", "ProducerName-Test")))
+            .params(getMockRequestParamsGetEservicesProducers("2", "0", "ProducerName")))
         .andReturn().getResponse();
 
     assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
     assertThat(response.getContentAsString()).isNotEmpty();
-    assertThat(response.getContentAsString()).contains("label");
-    assertThat(response.getContentAsString()).contains("value");
+    assertThat(response.getContentAsString()).contains("content");
   }
 
   @Test
   @DisplayName("given a valid producer name with no matching records, then returns an empty list")
   void testGetEservicesProducers_whenGivenValidProducerName_thenReturnsSearchProducerNameResponseListEmpty()
       throws Exception {
-    Mockito.when(service.getEservicesProducers(2, 0, "ProducerName-Test"))
-        .thenReturn(new ArrayList<Producer>());
+
+    producerExpectedList = SearchProducerNameResponse.builder().content(List.of()).build();
+
+    Mockito.when(service.getEservicesProducers(2, 0, "ProducerName-Test-1"))
+        .thenReturn(producerExpectedList);
+
+    MockHttpServletResponse response = mockMvc
+        .perform(get(apiGetEservicesProducersUrl)
+            .params(getMockRequestParamsGetEservicesProducers("2", "0", "ProducerName-Test-1")))
+        .andReturn().getResponse();
+    assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
+    assertThat(response.getContentAsString()).contains("[]");
+  }
+
+  @Test
+  @DisplayName("given a valid producer name with no matching records, then returns an empty object")
+  void testGetEservicesProducers_whenGivenValidProducerName_thenReturnsSearchProducerNameResponseNull()
+      throws Exception {
+    Mockito.when(service.getEservicesProducers(2, 0, "ProducerName-Test-1"))
+        .thenReturn(SearchProducerNameResponse.builder().build());
     MockHttpServletResponse response = mockMvc
         .perform(get(apiGetEservicesProducersUrl)
-            .params(getMockRequestParamsGetEservicesProducers("2", "0", "ProducerName-Test")))
+            .params(getMockRequestParamsGetEservicesProducers("2", "0", "ProducerName-Test-1")))
         .andReturn().getResponse();
     assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
-    assertThat(response.getContentAsString().equals("[]"));
+    assertThat(response.getContentAsString()).contains("null");
```

Comment:
why is this test the same as the one before but returning a different value?

Anyway, the empty response of a list endpoint should always be an empty list

Reply:
In the first test, the expected response is an empty list ([]) of the SearchProducerNameResponse object:
`producerExpectedList = SearchProducerNameResponse.builder().content(List.of()).build();`

In the second test, the expected response is the SearchProducerNameResponse object that is null:
```
     Mockito.when(service.getEservicesProducers(2, 0, "ProducerName-Test-1"))
         .thenReturn(SearchProducerNameResponse.builder().build());
```

For me they are 2 tests that verify different behaviors and answers, but, if it doesn't make sense to you, I'll refactor it